util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java

```diff
@@ -160,24 +173,39 @@ private void exportDataForAccountAndTable(final DatabaseExportOutputStream out,
             return;
         }
 
-        // Build the query - make sure to filter by account and tenant!
-        queryBuilder.append(" from ")
-                    .append(tableName)
-                    .append(" where ")
-                    .append(tableType.getAccountRecordIdColumnName())
-                    .append(" = :accountRecordId and ")
-                    .append(tableType.getTenantRecordIdColumnName())
-                    .append("  = :tenantRecordId");
+        if (tableType == TableType.EXTRA) {
+            queryBuilder.append(" from ")
+                        .append(tableName)
+                        .append(" where ")
+                        .append(tableType.getTenantRecordIdColumnName())
+                        .append("  = :tenantRecordId and (")
+                        .append(tableType.getAccountRecordIdColumnName())
+                        .append(" = :accountRecordId OR ")
+                        .append(tableType.getAccountRecordIdColumnName()) //TODO_354 - Custom logic for aviate_catalog, to include tenant level entries when accountId is null
+                        .append(" is null)")
+            ;
+
+        } else {
+
+            // Build the query - make sure to filter by account and tenant!
+            queryBuilder.append(" from ")
+                        .append(tableName)
+                        .append(" where ")
+                        .append(tableType.getAccountRecordIdColumnName())
+                        .append(" = :accountRecordId and ")
+                        .append(tableType.getTenantRecordIdColumnName())
+                        .append("  = :tenantRecordId");
+        }
 
         // Notify the stream that we're about to write data for a different table
         out.newTable(tableName, columnsForTable);
-
+        final TableType finalTableType = tableType;
         dbi.withHandle(new HandleCallback<Void>() {
             @Override
             public Void withHandle(final Handle handle) throws Exception {
                 final ResultIterator<Map<String, Object>> iterator = handle.createQuery(queryBuilder.toString())
-                                                                           .bind("accountRecordId", context.getAccountRecordId())
-                                                                           .bind("tenantRecordId", context.getTenantRecordId())
+                                                                           .bind("accountRecordId", finalTableType == TableType.EXTRA ? accountId : context.getAccountRecordId())
+                                                                           .bind("tenantRecordId", finalTableType == TableType.EXTRA ? tenantId : context.getTenantRecordId())
```

Comment:
This doesn't sound right: `tenantId` vs `tenantRecordId`?

Reply:
`tenantRecordId` is just the name of the parameter, the actual column name is `tenant_id` in case of the `aviate_catalog_xxx` tables. See [this line](https://github.com/reshmabidikar/killbill/blob/6465cc34592e4edaf9c14568e6d2d277f4c5fa4f/util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java#L76). See also [this comment](https://github.com/killbill/killbill-aviate-plugin/issues/354#issuecomment-2659996778) and [this slack thread](https://killbillio.slack.com/archives/G01C2P3V8G7/p1739799878262709).